http/http_exception.cc

```diff
@@ -24,6 +24,14 @@ void rethrowHttpException(int httpCode, const Utf8String & message, Any details)
         details2["context"]["details"] = jsonEncode(http.details);
         details2["context"]["error"] = http.message;
         throw HttpReturnException(httpCode == -1 ? http.httpCode : httpCode, message, details2);
+    } catch (const std::bad_alloc & exc) {
```

Comment:
Here we're rethrowing an exception once all unwinding has been done, in order to determine the type of the current exception.  It would be a very large amount of work to ensure that we could do good error handling and reporting without allocating any memory.  This solution is a 99% solution and the other 1% will be so much work I don't think it's worth doing.  (And this patch doesn't make things any worse as compared to without it).